bugbug/model.py

```diff
@@ -567,22 +569,55 @@ def train(self, importance_cutoff=0.15, limit=None):
 
             self.clf.fit(X_train, self.le.transform(y_train))
 
-        with open(self.__class__.__name__.lower(), "wb") as f:
-            pickle.dump(self, f, protocol=pickle.HIGHEST_PROTOCOL)
+        model_name = self.__class__.__name__.lower()
+
+        if hasattr(self.clf, "save_model"):
+            # If the model has the "save_model" attribute (is an XGBoost model),
+            # To handle multiple model files as one, place them in a directory with the expected model file name.
+            # when compressing the directory, it results in a single file.
+            os.makedirs(model_name, exist_ok=True)
+
+            # Save the XGBoost model to a file named "model.xgb"
+            self.clf.save_model(os.path.join(model_name, "model.xgb"))
```

Comment:
To allow us in future to support saving different types of models in the same way, we could specify the model type in the file name (e.g., `xgboost.ubj`).

Using `.ubj` as a file extension here will instruct XGBoost to save the model in binary JSON format.

Reply:
I made some updates 

https://github.com/mozilla/bugbug/blob/ca616d2c3fc95b218b892b8252f677b5b2eba2e4/bugbug/model.py#L580